src/explorer/trialApp/TrialAppTreeItem.ts

```diff
@@ -51,6 +52,20 @@ export class TrialAppTreeItem extends SiteTreeItemBase implements ISiteTreeItem
         this._tutorialNode = new GenericTreeItem(this, { label: 'Show tutorial', commandId: 'appService.ShowTutorial', contextValue: 'showTutorial', iconPath: getThemedIconPath('book') });
         this.logFilesNode = new LogFilesTreeItem(this, this.client);
         this.deploymentsNode = new TrialAppDeploymentsTreeItem(this, this.client, {}, {});
+
+        // seconds * ms
+        const interval: number = 60 * 1000;
+        const intervalId: NodeJS.Timeout = setInterval(
+            async () => {
+                if (ext.azureAccountTreeItem.trialAppNode !== this || this.client.isExpired) {
```

Comment:
I worry we could get stuck in a weird loop if we refresh the account tree item every time `ext.azureAccountTreeItem.trialAppNode !== this`. For example:

1. TrialAppTreeItem1 is created and sets interval
2. User refreshes (creating TrialAppTreeItem2)
3. TrialAppTreeItem1 detects the new instance, refreshes the account tree item (creating TrialAppTreeItem3), and clears it's interval
3. TrialAppTreeItem2 detects the new instance, refreshes the account tree item (creating TrialAppTreeItem4), and clears it's interval
4. etc.

There's probably a lot of ways to refactor the `if` conditions, but here's one way that doesn't require checking the same condition twice:
```
if (this.client.isExpired) {
    await ext.azureAccountTreeItem.refresh();
} else if (ext.azureAccountTreeItem.trialAppNode === this) {
    await this.refresh();
    return;
}

clearInterval(intervalId);
```